pkg/controllers/gpu/controller.go

```diff
@@ -170,6 +171,36 @@ func (c *controller) NewDaemonSet(rawObj runtime.Object) *apps.DaemonSet {
 			},
 		},
 	}
+	// add the optional InitImage
+	if devicePlugin.Spec.InitImage != "" {
+		daemonSet.Spec.Template.Spec.InitContainers = []v1.Container{
+			{
+				Image:           devicePlugin.Spec.InitImage,
+				ImagePullPolicy: "IfNotPresent",
+				Name:            "intel-gpu-initcontainer",
+				SecurityContext: &v1.SecurityContext{
+					ReadOnlyRootFilesystem: &yes,
+				},
+				VolumeMounts: []v1.VolumeMount{
+					{
+						MountPath: "/etc/kubernetes/node-feature-discovery/source.d/",
+						Name:      "nfd-source-hooks",
+					},
+				},
+			},
+		}
+		daemonSet.Spec.Template.Spec.Volumes = append(daemonSet.Spec.Template.Spec.Volumes,
+			v1.Volume{
+				Name: "nfd-source-hooks",
+				VolumeSource: v1.VolumeSource{
+					HostPath: &v1.HostPathVolumeSource{
+						Path: "/etc/kubernetes/node-feature-discovery/source.d/",
+						Type: &directoryOrCreate,
+					},
+				},
+			})
+	}
+	return &daemonSet
 }
 
 func (c *controller) UpdateDaemonSet(rawObj runtime.Object, ds *apps.DaemonSet) (updated bool) {
```

Comment:
I tested this PR and noticed the updates wont work. This func needs changes still.

Reply:
Maybe now?

Reply:
@uniemimu this works now: tested CR creation without `initImage`, then added `initImage` (reconcile OK) and finally removed `initImage` (reconcile also OK). Can you move this 'ready for review'